pkg/brokerapi/fake/fake.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fake
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	uuid "github.com/satori/go.uuid"
+)
+
+// Client implements a fake broker API client, useful for unit testing. None of the methods on
+// the client are concurrency-safe
+type Client struct {
+	CatalogClient
+	InstanceClient
+	BindingClient
```

Comment:
Oh, is this why you used the word "clients" above? Do we expect people to view these as 3 different clients?  This might imply some kind of architectural design point that I'm not sure we have.